### ISSUE TITLE
Plugin broken after upgrade to ESLint 6.0.0

### ISSUE THREAD

**Issue by Baune8D** [vuejs/eslint-plugin-vue#911]
**Tell us about your environment**	
* **ESLint version:** 6.0.0
* **eslint-plugin-vue version:** 5.2.2
* **Node version:**	 10.15.3

**Please show your full configuration:**

```javascript
module.exports = {
  env: {
    browser: true,
    commonjs: true,
    es6: true,
    node: true,
    jquery: true,
  },
  extends: [
    'eslint:recommended',
    'airbnb-base',
    'plugin:vue/recommended',
    'plugin:prettier/recommended',
    'prettier/vue',
  ],
  parser: 'vue-eslint-parser',
  parserOptions: {
    parser: 'babel-eslint',
    babelOptions: {
      configFile: require.resolve(
        `@<package>/build/configs/babel.config.vue.js`,
      ),
    },
  },
  overrides: [
    {
      files: ['**/*.ts', '**/*.vue'],
      extends: [
        'eslint:recommended',
        'airbnb-base',
        'plugin:@typescript-eslint/eslint-recommended',
        'plugin:@typescript-eslint/recommended',
        'plugin:vue/recommended',
        'plugin:prettier/recommended',
        'prettier/@typescript-eslint',
        'prettier/vue',
      ],
      parser: 'vue-eslint-parser',
      parserOptions: {
        parser: '@typescript-eslint/parser',
        project: path.resolve(<tsconfigPath>, 'tsconfig.json'),
        extraFileExtensions: ['.vue'],
      },
    },
  ],
  settings: {
    'import/resolver': {
      typescript: {
        directory: <tsconfigPath>,
      },
      webpack: {
        config: {
          resolve: {
            alias: <aliases>,
            extensions: ['.js', '.ts, '.vue'],
          },
        },
      },
    },
  },
};
```

**What did you do?**	
Upgrade ESLint to 6.0.0.

Problem occurs here:
https://github.com/vuejs/eslint-plugin-vue/blob/ae03c281a0bf031f65ee8976b3a3228053e61ca2/lib/rules/order-in-components.js#L8

This seems to be the same basic issue as here:
https://github.com/typescript-eslint/typescript-eslint/issues/563

[Comment](https://github.com/typescript-eslint/typescript-eslint/issues/563#issuecomment-502183156)
> ```traverser.js``` has been moved from ```lib/util/traverser.js``` to ```lib/shared/traverser.js```
in commit [eslint#219aecb](https://github.com/eslint/eslint/commit/219aecb78bc646d44bad27dc775a9b3d3dc58232) PR [eslint#11555](https://github.com/eslint/eslint/pull/11555)

**What did you expect to happen?**
Lint successfully

**What actually happened?**
Error: Failed to load plugin 'vue' declared in 'Assets/.eslintrc.js': Cannot find module 'eslint/lib/util/traverser'
Referenced from: /\<path>/Assets/.eslintrc.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
    at Function.Module._load (internal/modules/cjs/loader.js:508:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/\<path>/node_modules/eslint-plugin-vue/lib/rules/order-in-components.js:8:19)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)

**Comment by vesper8** [comment]
similar problem. I am using "latest" on many deps in my package.json and today I started getting this error

```
Module build failed (from ./node_modules/@vue/cli-plugin-eslint/node_modules/eslint-loader/index.js):
Error: Cannot find module 'eslint/lib/formatters/stylish'
```

during yarn install:

```
eslint-plugin-vue > vue-eslint-parser@5.0.0" has unmet peer dependency "eslint@^5.0.0
```

Specifically, this package.json was working fine before today:

```
  "devDependencies": {
    "@babel/core": "latest",
    "@vue/cli-plugin-babel": "next",
    "@vue/cli-plugin-e2e-cypress": "next",
    "@vue/cli-plugin-eslint": "next",
    "@vue/cli-plugin-pwa": "next",
    "@vue/cli-plugin-unit-mocha": "next",
    "@vue/cli-service": "next",
    "@vue/eslint-config-airbnb": "latest",
    "@vue/test-utils": "latest",
    "chai": "latest",
    "eslint": "latest",
    "eslint-plugin-vue": "latest",
    "sass": "latest",
    "sass-loader": "latest",
    "vue-template-compiler": "latest"
  }
```

and as of today it isn't working anymore

I know using next/latest is not the best idea but this is only for development purposes

**Comment by Baune8D** [comment]
There seems to be a PR regarding this issue here: https://github.com/vuejs/eslint-plugin-vue/pull/910

**Comment by yi-ge** [comment]
I have the same problem.

**Comment by guastallaigor** [comment]
Same problem here, any workaround is appreciated.

**Comment by shlima** [comment]
+1

**Comment by Baune8D** [comment]
This issue seems to be resolved in v5.2.3
